### PR TITLE
add more border styles to BorderSide style element

### PR DESCRIPTION
**Pull request by seekerquest**
Add dashed", "dotted", "double", "groove", "ridge", "inset", "outset" to BorderSide class

### CODE REVIEW

File: mesop/component_helpers/style.py [hunk 1]
```diff
@@ -73,12 +73,14 @@ class BorderSide:
                                 a string with a unit (e.g., '2em'), or None for no width.
       color: The color of the border, represented as a string. This can be any valid CSS color value,
                           or None for no color.
-      style: The style of the border, which can be 'none' for no border, 'solid' for a solid line.
+      style: The style of the border, which can be 'none' for no border, 'solid' for a solid line, 'dashed' for dashed line,
```
Comment (richard-to) [inline mesop/component_helpers/style.py:76]: Maybe you can link to this as well: https://developer.mozilla.org/en-US/docs/Web/CSS/border-style

File: mesop/component_helpers/style.py [hunk 2]
```diff
@@ -73,12 +73,14 @@ class BorderSide:
...
+              'dotted' for dotted line, 'double' for double line, 'groove' for 3D grooved border, 'ridge' for 3D ridged border,
+              'inset' for 3D inset border, 'outset' for 3D outset border.
   """
 
   width: int | str | None = None
   color: str | None = None
-  style: Literal["none", "solid"] | None = None
+  style: Literal["none", "solid", "dashed", "dotted", "double", "groove", "ridge", "inset", "outset"] | None = None
```
Comment (richard-to) [inline mesop/component_helpers/style.py:83]: Seems like there is also a "hidden" style, which seems slightly different than none. So we may want to add it for completeness.

**Review by richard-to — CHANGES_REQUESTED** [review]
Can you also update "box" demo example to use some of the added styles? That's just to make sure we have an example of it working. See: https://github.com/google/mesop/blob/main/demo/box.py

--------

Also looks like someone else also posted a pull request for the border side change after you: https://github.com/google/mesop/pull/509. Looks like what happened is that they didn't see your message. And I had referred that person to this issue from another issue.

Since you were first, I will give your pull request precedence.

**Review by richard-to — APPROVED** [review]
LGTM - Thanks

**Comment by richard-to** [comment]
Looks like the pre-submit checks failed. Just some linting stuff. The best option is to set up the pre-commits described here: https://google.github.io/mesop/internal/development/

The precommits should manually run the ruff formatter to fix the formatting.

For a manual run, I think the command is something like:

```
pre-commit run --files FILES
```

**Comment by seekerquest** [comment]
Thanks for sharing the steps to fix the pre-checks. I think this has now passed the pre-checks.

**Comment by richard-to** [comment]
Hmm looks like there is another type checking issue on the demo page. It's an interesting one. I'm not quite sure how best to resolve that type checking issue.

For simplicity, you may just want to remove the loop. Then for the styles, just hardcode to different styles.

**Comment by richard-to** [comment]
Closing this one. We ended up merging a similar pull request in https://github.com/google/mesop/pull/509

**Pull request closed without merging**